<a href="https://colab.research.google.com/github/amandabenjamin/Predicting-Strokes/blob/main/Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Amanda Benjamin

Date: 4/08/2022

# Dataset 1: Stroke Prediction Dataset

**1) Source of the data**

I found this dataset on Kaggle, you can access the dataset from the following link: https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset. 

This data is available for educational use so long as the author is credited. 

**2) Brief description of data**

This dataset allows one to predict the likelyhood of a stroke given relavent information about each unique patient including but not limited to: gender, age, marriage status, average glucose level, and bmi. 

**3) What is the target?**

The target is the 'stroke' outcome (column) - where 1 indicates the patient had a stroke and 0 indicates the patient did not have a stroke. 

**4) Is this a classification or regression problem?**

This is a classification problem. 

**5) How many features?**

There are 11 clinicnal features that will be used to predict a 'stroke outcome'. 

**6) How many rows of data?**

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
import pandas as pd
df = pd.read_csv('/content/drive/MyDrive/Coding Dojo/07 Week 7: Classification Models/stroke_prediction.csv')
df.shaped

There are 5110 rows of data (including the header). 

**7) What, if any, challenges do you foresee in cleaning, expoloring, or modeling with this dataset?**

I see there is a note for unknown smoker status - potentially, there might be too many nan values for this feature to drop without affecting the dataset/model outcomes. 

# Dataset 2: 

**1) Source of data**

I found this dataset on Kaggle, you can access the dataset from the following link: https://www.kaggle.com/datasets/uciml/red-wine-quality-cortez-et-al-2009

This data is available with an open public license. 

**2) Brief description of data**

This dataset provides a collection of physicochemical and sensory variables related to red and white variants of 'Vinho Verde' wine used to predict the 'rating' of the wine. 

**3) What is the target?**

The target is the quality score of the wine (between 0 and 10). 

**4) Is this a classification or regression problem?**

This is a regression problem. 

**5) How many features?**

There are ll features - including but not limited to: the density of the wine, pH, and acidity (fixed and volatile). 

**6) How many rows of data?**

In [7]:
import pandas as pd
df = pd.read_csv('/content/drive/MyDrive/Coding Dojo/07 Week 7: Classification Models/winequality-red.csv')
df.shape

(1599, 12)

There are 1599 rows of data (including the header). 

**7) What, if any, challenges do you foresee in cleaning, expoloring, or modeling with this dataset?**

The visualizations might be limited for this dataset may be limited as there is only numerial data provided for this dataset - potentially making interpretation hard for me and my audience. 